Hunting Queries/MultipleDataSources/RareDNSLookupWithDataTransfer.txt

```diff
@@ -0,0 +1,96 @@
+// Name: Rare DNS lookups today and data transfer to/from those domains - RareDNSLookupWithDataTransfer
+// Description: This query is designed to help identify rare DNS connections and resulting data transfer to/from the associated domain.
+// This can help identify unexpected large data transfers to or from internal systems which may indicate data exfil or malicious tool download.
+// Feel free to add additional data sources to connect DNS results too various network data that has byte transfer information included.
+//
+// Id: 06c52a66-fffe-4d3b-a05a-646ff65b7ec2
+//
+// DataSource:#DnsEvents, #CommonSecurityLog, #WireData, #VMConnection
+//
+// Tactics: #Exfiltration, #C2
+//
+let lookbackint = 7;
+let lookupThreshold = lookbackint*3;
+let lookbackstring = strcat(tostring(lookbackint),".00:00:00");
+let lookbacktime = totimespan(lookbackstring)+1d;
+let starttime = startofday(ago(lookbacktime)); //startofday is setting to 00:00:00 for the given days ago
+let endtime = startofday(now(-1d)); 
+let binvalue = 1;
+let bintime = make_timespan(binvalue,0);
+let avgCalc = starttime/1h;
+let PrivateIPregex = @'^127\.|^10\.|^172\.1[6-9]\.|^172\.2[0-9]\.|^172\.3[0-1]\.|^192\.168\.'; 
+// Identify all domain lookups
+let DomainLookups = DnsEvents 
+| where TimeGenerated >= starttime and TimeGenerated <= endtime 
+| where SubType == "LookupQuery"
+| where isnotempty(IPAddresses)
+| extend Domain = iff(countof(Name,'.') >= 2, strcat(split(Name,'.')[-2], '.',split(Name,'.')[-1]), Name)
+| summarize count() by Domain
```

Comment:
let's avoid implicit names when we can? DomainCount=count() ?

Reply:
Changed - 
| summarize DomainCount = count() by Domain
| project Domain, DailyAvgLookupCountOverLookback = DomainCount/lookbackint;